sdk/src/Services/DynamoDBv2/Custom/DataModel/BaseOperationConfig.cs

```diff
@@ -0,0 +1,100 @@
+ *  http://aws.amazon.com/apache2.0
+ * 
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+namespace Amazon.DynamoDBv2.DataModel
+{
+    /// <summary>
+    /// Base class for operation-specific configurations for DynamoDB object persistence operations.
+    /// </summary>
+    /// <remarks>
+    /// This should only contain members that are relevant to all object persistence operations, 
+    /// anything operation-specific should be added to derived classes.
+    /// </remarks>
+#if NET8_0_OR_GREATER
+    [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCode(Custom.Internal.InternalConstants.RequiresUnreferencedCodeMessage)]
+#endif
+    public abstract class BaseOperationConfig
+    {
+        /// <summary>
+        /// Indicates which DynamoDB table to use. This overrides the table specified 
+        /// by the <see cref="DynamoDBTableAttribute"/> on the .NET objects that you're saving or loading.
+        /// </summary>
+        public string OverrideTableName { get; set; }
+
+        /// <summary>
+        /// Directs <see cref="DynamoDBContext"/> to prefix all table names
+        /// with a specific string. If this is null or empty, no prefix is used 
+        /// and default table names are used.
+        /// </summary>
+        public string TableNamePrefix { get; set; }
+
+        /// <summary>
+        /// The object persistence model API relies on an internal cache of the DynamoDB table's metadata to construct and validate 
+        /// requests. This controls how the cache key is derived, which influences when the SDK will call
+        /// IAmazonDynamoDB.DescribeTable(string) internally to populate the cache.
+        /// </summary>
+        /// <remarks>
+        /// For <see cref="MetadataCachingMode.Default"/> the cache key will be a combination of the table name, credentials, region and service URL. 
+        /// For <see cref="MetadataCachingMode.TableNameOnly"/> the cache key will only consist of the table name. This reduces cache misses in contexts
+        /// where you are accessing tables with identical structure but using different credentials or endpoints (such as a multi-tenant application).
+        /// </remarks>
+        public MetadataCachingMode? MetadataCachingMode { get; set; }
+
+        /// <summary>
+        /// If true disables fetching table metadata automatically from DynamoDB. Table metadata must be 
+        /// defined by <see cref="DynamoDBAttribute"/> attributes and/or in <see cref = "AWSConfigsDynamoDB"/>.
+        /// </summary>
+        /// <remarks>
+        /// Setting this to true can avoid latency and thread starvation due to blocking asynchronous 
+        /// IAmazonDynamoDB.DescribeTable(string) calls that are used to populate the SDK's cache of 
```

Comment:
Can you also wrap `IAmazonDynamoDB.DescribeTable(string)` in a `cref`

Reply:
It's not available in all targets since `DescribeTable` vs. `DescribeTableAsync` vary. 
The API reference generator doesn't generate target-specific pages, so I don't think it's worth `#if` inside the comment block. And technically we're always calling an _internal_ `DescribeTable` even from async targets, which is the root cause of the sync-over-async issues.

Reply:
I switched to just a plain `DescribeTable` in https://github.com/aws/aws-sdk-net/pull/3380/commits/3e03c1ee9c64a1b54987c0e37acbb7e4309b5314 since this keeps coming up. 

From the user's perspective, the important part is that we're calling that API implicitly. The exact invocation is less important.